python/lsst/analysis/tools/tasks/sourceObjectTableAnalysis.py

```diff
@@ -435,10 +436,11 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
         for refCatRef in inputs["refCat"]:
             refCat = refCatRef.get(
                 parameters={
-                    "columns": ["detect_isPrimary", self.config.ra_column, self.config.dec_column]
+                    "columns": ["detect_isPrimary", self.config.ra_column, self.config.dec_column, "objectId"]
                     + refCatSelectors
                 }
             )
+            refCat.set_index("objectId")
```

Comment:
Is this setting an astropy table index?

Reply:
No this one's still a `DataFrame`, because there was a lot of `pandas`-heavy code using it.  I'm setting the index here because the thing that's still being read in as a `DataFrame` is now being written as `ArrowAstropy` and hence isn't getting the index set on write, as the code here previously expected.